aten/src/ATen/cpu/vec256/vec256_complex_double_no_vec.h

```diff
@@ -0,0 +1,328 @@
+      }
+    }
+    return vec;
+  }
+  static Vec256<std::complex<double>> loadu(const void* ptr, int64_t count = size()) {
+    Vec256 vec;
+    std::memcpy(vec.values, ptr, count * sizeof(std::complex<double>));
+    return vec;
+  }
+  void store(void* ptr, int count = size()) const {
+    std::memcpy(ptr, values, count * sizeof(std::complex<double>));
+  }
+  const std::complex<double>& operator[](int idx) const {
+    return values[idx];
+  }
+  std::complex<double>& operator[](int idx) {
+    return values[idx];
+  }
+  Vec256<std::complex<double>> map(std::complex<double> (*f)(const std::complex<double> &)) const {
+    Vec256<std::complex<double>> ret;
+    for (int i = 0; i != size(); i++) {
+      ret[i] = f(values[i]);
+    }
+    return ret;
+  }
+  Vec256<std::complex<double>> abs() const {
+    Vec256<std::complex<double>> ret;
+    for (int i = 0; i < size(); i++) {
+      ret[i] = std::abs(values[i]);
+    }
+    return ret;
+  }
+  Vec256<std::complex<double>> real() const {
+    Vec256<std::complex<double>> ret;
+    for (int i = 0; i < size(); i++) {
+      ret[i] = std::real(values[i]);
+    }
+    return ret;
+  }
+  Vec256<std::complex<double>> imag() const {
+    Vec256<std::complex<double>> ret;
+    for (int i = 0; i < size(); i++) {
+      ret[i] = std::imag(values[i]);
+    }
+    return ret;
+  }
+  Vec256<std::complex<double>> acos() const {
+    return map(std::acos);
+  }
+  Vec256<std::complex<double>> asin() const {
+    return map(std::asin);
+  }
+  Vec256<std::complex<double>> atan() const {
+    return map(std::atan);
+  }
+  Vec256<std::complex<double>> atan2(const Vec256<std::complex<double>> &b) const {
+    AT_ERROR("not supported for complex numbers");
+  }
+  Vec256<std::complex<double>> erf() const {
+    AT_ERROR("not supported for complex numbers");
+  }
+  Vec256<std::complex<double>> erfc() const {
+    AT_ERROR("not supported for complex numbers");
+  }
+  Vec256<std::complex<double>> exp() const {
+    return map(std::exp);
+  }
+  Vec256<std::complex<double>> expm1() const {
+    AT_ERROR("not supported for complex numbers");
+  }
+  Vec256<std::complex<double>> log() const {
+    return map(std::log);
+  }
+  Vec256<std::complex<double>> log2() const {
+    AT_ERROR("not supported for complex numbers");
+  }
+  Vec256<std::complex<double>> log10() const {
+    return map(std::log10);
+  }
+  Vec256<std::complex<double>> log1p() const {
+    AT_ERROR("not supported for complex numbers");
+  }
+  Vec256<std::complex<double>> sin() const {
+    return map(std::sin);
+  }
+  Vec256<std::complex<double>> sinh() const {
+    return map(std::sinh);
+  }
+  Vec256<std::complex<double>> cos() const {
+    return map(std::cos);
+  }
+  Vec256<std::complex<double>> cosh() const {
+    return map(std::cosh);
+  }
+  Vec256<std::complex<double>> ceil() const {
+    return map([](const std::complex<double> &x) {
+      return std::complex<double>(std::ceil(std::real(x)), std::ceil(std::imag(x)));
```

Comment:
...really?

Reply:
I know, these are the easy functions to vectorize. I was planning to do that next week.

Reply:
I am actually talking more semantics: should `ceil` on complex really be elementwise? (I don't know the answer to this question!)

Reply:
Yes, I would have to profile a couple approaches to see which is better. I agree that vec256_complex_double_no_vec.h and vec256_complex_float_no_vec.h need lots of work before they should be put in.